cmd/rollout-operator/main.go

```diff
@@ -87,6 +103,32 @@ func (cfg config) validate() error {
 	if (cfg.kubeAPIURL == "") != (cfg.kubeConfigFile == "") {
 		return errors.New("either configure both Kubernetes API URL and config file or none of them")
 	}
+	if cfg.useZoneTracker {
+		if cfg.objectStorageProvider == "" {
+			return errors.New("the object storage provider for the zone tracker has not been specified")
+		}
+		if cfg.bucketName == "" {
+			return errors.New("the bucket name for the zone tracker has not been specified")
+		}
+		supportedProviders := []string{"s3", "gcs", "azure"}
+		supported := false
+		for _, provider := range supportedProviders {
+			if cfg.objectStorageProvider == provider {
+				supported = true
+				break
+			}
+		}
+		if !supported {
+			return fmt.Errorf("the provided object storage provider is not supported (currently supported providers: %s)", strings.Join(supportedProviders, ", "))
+		}
+		if cfg.objectStorageProvider == "azure" && (cfg.accountName == "" || cfg.objectStorageEndpoint == "") {
+			return errors.New("the account name for the Azure object storage provider has not been specified")
+		}
+		if cfg.objectStorageProvider == "s3" && (cfg.objectStorageEndpoint == "" || cfg.objectStorageRegion == "") {
+			return errors.New("the endpoint and region for the S3 object storage provider have not been specified")
+		}
```

Comment:
Do we need a similar check for GCS?

Reply:
GCS only requires the bucket to be specified, which is checked above this conditional